### PR TITLE
First README version

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I've added the setup and usage guidelines to the README page.

@reviewbybees @jtnord @gusrieber

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,103 @@
-# Docker Traceability Plugin
+# CloudBees Docker Traceability Plugin
```
Comment (jtnord) [inline README.md]: no "plugin"

File: README.md [hunk 2]
```diff
@@ -1,3 +1,103 @@
...
+ * Support of search queries
+ 
+ ![Main page](/doc/images/root-action.png)
+
+#Installation Guidelines
+
+##Plugin setup
+
+1. Install Docker Traceability plugin from Jenkins Update Center
+2. If you are going to use Docker Traceability for images built on Jenkins, consider the installation of other
```
Comment (jtnord) [inline README.md]: of other... what?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,103 @@
...
+3. Configure the security. This step is **very important**, because the plugin can store raw JSON, which may contain sensitive info like passwords
```
Comment (jtnord) [inline README.md]: its not **the** security, just security.  But you probably need to link to a wiki article as this is Jenkins security not security of something else.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,103 @@
...
+ * Data polling from Jenkins (including Docker API-compatible JSONs)
+ * Support of search queries
+ 
+ ![Main page](/doc/images/root-action.png)
+
+#Installation Guidelines
+
+##Plugin setup
+
+1. Install Docker Traceability plugin from Jenkins Update Center
```
Comment (jtnord) [inline README.md]: do we need to state this? it should be pretty obvious.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,103 @@
...
+2. If you are going to use Docker Traceability for images built on Jenkins, consider the installation of other
+3. Configure the security. This step is **very important**, because the plugin can store raw JSON, which may contain sensitive info like passwords
+ * The plugin introduces new permissions, which allow to restrict the access
+ * **Read** - Allows to retrieve details like full container/info dumps.
```
Comment (jtnord) [inline README.md]: these permissions are scoped (I hope) so the scope needs to be mentioned.

File: README.md [hunk 6]
```diff
@@ -1,3 +1,103 @@
...
+
+##Plugin setup
+
+1. Install Docker Traceability plugin from Jenkins Update Center
+2. If you are going to use Docker Traceability for images built on Jenkins, consider the installation of other
+3. Configure the security. This step is **very important**, because the plugin can store raw JSON, which may contain sensitive info like passwords
+ * The plugin introduces new permissions, which allow to restrict the access
+ * **Read** - Allows to retrieve details like full container/info dumps. 
+      Web interfaces are being managed by common Jenkins Read permission
+ * **Submit** - Allows to submit deployment records from [API](#API)
```
Comment (jtnord) [inline README.md]: via **the** remote api.  (I assume its the remote API rather than the internal java API)

File: README.md (hunk 2 above)
Comment (oleg-nenashev) [inline README.md]: I was going to add links to plugins with a supported integration, but I've forgot about it

File: README.md [hunk 7]
```diff
@@ -1,3 +1,103 @@
...
+ * **Delete** - Allows to delete deployment records or entire fingerprints
```
Comment (jtnord) [inline README.md]: allows **the deletion** of deployment...

File: README.md [hunk 8]
```diff
@@ -1,3 +1,103 @@
...
+
+```
+The section is under construction
+```
+
+#Use-cases
+
+##Submitting deployment records
+
+Currently the plugin does not support an automatic polling of events from external Docker servers. The events should be submitted by external clients or other Jenkins plugins.
```
Comment (jtnord) [inline README.md]: I wouldn't say currently - just that it doesn't do it.

File: README.md [hunk 9]
```diff
@@ -1,3 +1,103 @@
...
+## Getting info from the plugin
+
+For each container record the plugin publishes the info on the container summary page. A summary status is being also added to the parent image page.
+
+![Docker container page](/doc/images/container-page.png)
+
+![Docker image page](/doc/images/image-page.png)
+
+
+If an external client submits an info about the image (can be retrieved using *docker inspect imageId* command), Docker Traceability Plugin captures this image and adds a new facet to the image fingerprint page.
```
Comment (jtnord) [inline README.md]: submits information about...    (which can be retreived....)

File: README.md (hunk 1 above)
Comment (oleg-nenashev) [inline README.md]: Yes, it makes sense if we're planning to add client-side, Docker demos and other such stuff later

File: README.md (hunk 5 above)
Comment (oleg-nenashev) [inline README.md]: Scope=Jenkins, Group="Docker Traceability"

File: README.md (hunk 8 above)
Comment (oleg-nenashev) [inline README.md]: Agreed. BTW, it may be implemented later by this plugin or another one using its extension points

File: README.md [hunk 10]
```diff
@@ -1,3 +1,103 @@
...
+ * Support of search queries
+ 
+ ![Main page](/doc/images/root-action.png)
+
+#Installation Guidelines
+
+##Plugin setup
+
+1. Install CloudBees Docker Traceability plugin from Jenkins Update Center
+2. Install other Jenkins plugins, which produce image fingerprints to be traced by the plugin (see [Integrations][#Integrations])
```
Comment (jtnord) [inline README.md]: wrong brackets for the #link should be ( )

File: README.md [hunk 11]
```diff
@@ -1,3 +1,103 @@
...
+The section is under construction
+```
+
+#Use-cases
+
+##Submitting deployment records
+
+The plugin does not support an automatic polling of events from external Docker servers. The events should be submitted by external clients or other Jenkins plugins.
+
+**Warning!** Currently the plugin accepts the info for previously registered fingerprints only. Other submissions will be ignored. Initial image records should be created by other plugins using (see [Integrations][#Integrations])
```
Comment (jtnord) [inline README.md]: wrong brackets for the #link should be ( )

File: README.md [hunk 12]
```diff
@@ -1,3 +1,103 @@
...
+
+Raw data is accessible via the [plugin's API](#API) or via hyperlinks on info pages.
+
+## Search
+
+You can search deployments by container IDs using the "Search" control on the "Docker Traceability" page. You can also query containers using the [plugin's API](#API).
+
+#Integrations
+
+CloudBees Docker Traceability plugin is based on fingerprints provided by [Docker Commons Plugin][docker-commons]. The plugin just adds additional  facets to main fingerprint pages, so any other plugin can contribute to the UI by adding additional facets to the fingerprint.
```
Comment (jtnord) [inline README.md]: wrong brackets for the link should be ( ) (and I don't think it's correct even then)

File: README.md [hunk 13]
```diff
@@ -1,3 +1,103 @@
...
+
+## Search
+
+You can search deployments by container IDs using the "Search" control on the "Docker Traceability" page. You can also query containers using the [plugin's API](#API).
+
+#Integrations
+
+CloudBees Docker Traceability plugin is based on fingerprints provided by [Docker Commons Plugin][docker-commons]. The plugin just adds additional  facets to main fingerprint pages, so any other plugin can contribute to the UI by adding additional facets to the fingerprint. 
+
+See [Docker Commons Plugin Wiki page][docker-commons] to get an info about existing fingerprint contributors.
```
Comment (jtnord) [inline README.md]: ditto () and link url

File: README.md (hunk 10 above)
Comment (orrc) [inline README.md]: On this topic, [grip](https://github.com/joeyespo/grip) is a really useful piece of software when writing larger pieces of documentation for use on GitHub.

Comment (oleg-nenashev) [inline README.md]: I use Atom with a Markdown preview, but the link seemed to be correct there. BTW it violates the format spec in any case

**Comment by oleg-nenashev** [comment]
@jtnord 
The failure is not related to the docs in any case.

**Comment by jtnord** [comment]
:+1: documentation is good.

**Comment by jglick** [comment]
:+1:

**Pull request merged**